Added support for the "Not" Operator (#195)
DataDog/datadog-sync-cli: README.md

- `Name`: Attribute key to filter on. This can be any attribute represented in dot notation (such as `attributes.user_count`). [required]
- `Value`: Regex to filter attribute value by. Note: special regex characters need to be escaped if filtering by raw string. [required]
- `Operator`: Available operators are below. All invalid operator's default to `ExactMatch`.

>   - `Not`: Inverse of `ExactMatch`.

Reviewer: ```suggestion
  - `Not`: Match not equal to `Value`.
```

>   - `SubString`: Sub string matching. (This operator will be removed in future releases, for more information read the [SubString and ExactMatch Deprecation Section](#substring-and-exactmatch-deprecation) below)
>   - `ExactMatch`: Exact string match. (This operator will be removed in future releases, for more information read the [SubString and ExactMatch Deprecation Section](#substring-and-exactmatch-deprecation) below)
>   - `Not`: Inverse of `ExactMatch`.

Reviewer: ```suggestion
  - `Not`: Inverse of `ExactMatch`.
  - `SubString` (*Deprecated*): Sub string matching. (This operator will be removed in future releases. See [SubString and ExactMatch Deprecation](#substring-and-exactmatch-deprecation)  section.)
  - `ExactMatch` (*Deprecated*): Exact string match. (This operator will be removed in future releases. See [SubString and ExactMatch Deprecation](#substring-and-exactmatch-deprecation)  section.)
```

By default, if multiple filters are passed for the same resource, `OR` logic is applied to the filters. This behavior can be adjusted using the `--filter-operator` option.

###### SubString and ExactMatch Deprecation

> In future releases the `SubString` and `ExactMatch` Operator will be removed. This is because the `Value` key supports regex so both of these scenarios are covered by just writing the appropriate regex.  Below is an example:

Reviewer: ```suggestion
In future releases the `SubString` and `ExactMatch` Operator will be removed as the `Value` key supports regex. Both, `SubString` and `ExactMatch`, can be replicated using the appropriate regex.  See example below for migrating existing usage of `ExactMatch` and `Substring` to `Value` regex:
```

> Let's take the scenario where you would like to filter for Monitors that have the `filter test` in the `name` attribute, below you can see how you would filter for both `SubString` and `ExactMatch`:

Reviewer: ```suggestion
Let's take the scenario where you would like to filter for monitors that have the `filter test` in the `name` attribute:
```

| Operator| Status | Command |
| :-: | :-: | :-: |
| SubString | Before  Deprecation | `--filter 'Type=monitors;Name=name;Value=filter test;Operator=SubString'` |
| Using `Value` | After Deprecation | `--filter 'Type=monitors;Name=name;Value=.*filter test.*` |
| ExactMatch | Before  Deprecation | `--filter 'Type=monitors;Name=name;Value=filter test;Operator=ExactMatch'` |

> | Using Value | After Deprecation | `--filter 'Type=monitors;Name=name;Value=^filter test$` |

Reviewer: Lets remove the status column here and use code blocks for operators and Value
